pkg/collector/jobevent_watcher.go

```diff
@@ -23,10 +23,21 @@ import (
 
 var watchStartTime = meta_v1.Now()
 
+var podFilter *regexp.Regexp
+
 type EventHandler struct {
 	collection *CronJobCollection
 }
 
+func createPodFilter() *regexp.Regexp {
+	if podFilter := viper.GetString("pod-filter"); podFilter != "" {
+		log.Debugf("pod filter enabled: %s", podFilter)
+		return regexp.MustCompile(podFilter)
```

Comment:
This is ok as-is but I would note that this will throw a panic if it fails, and that will probably show up to the user in an ugly way and without context. I'd probably suggest returning `err` instead so it can be passed to the CLI that the regex has an error